### PR TITLE
feat: sync AxonBlock, Metadata, Proof with axon

### PR DESCRIPTION
**Pull request by wenyuanhust**
**What this PR does / why we need it**:

This PR syncs with latest changes to AxonBlock, Metadata, Proof of Axon. Moreover, the corresponding deserialization methods to these structures are updated too.

**Which docs this PR relation**:

https://github.com/cryptape/layer2-internal/issues/406

### CODE REVIEW

**Review by ashuralyk — COMMENTED** [review]
can you use real data and interface from Axon repo to fulfill the examples to check the validation of interactivity with Axon ?

**Comment by wenyuanhust** [comment]
> can you use real data and interface from Axon repo to fulfill the examples to check the validation of interactivity with Axon ?

Yes, it will be updated. I'm trying to pass the mock data firstly

File: examples/verify_proof.rs [hunk 1]
```diff
@@ -7,28 +7,65 @@ fn read_json<T: DeserializeOwned>(path: &str) -> T {
...
+            proposer:                 Default::default(),
+            prev_state_root:          Default::default(),
+            transactions_root:        Default::default(),
+            signed_txs_hash:          Default::default(),
+            timestamp:                0,
+            number:                   100,
+            gas_limit:                Default::default(),
+            extra_data:               Default::default(),
+            base_fee_per_gas:         Default::default(),
+            proof:                    Proof::default(),
```
Comment (KaoImin) [inline examples/verify_proof.rs]: The mocked data should be generated by `rand`.

File: axon-tools/src/proof.rs [hunk 2]
```diff
@@ -52,12 +57,14 @@ pub fn verify_proof(
...
         block_hash: Bytes::from(proof.block_hash.0.to_vec()),
     };
+    println!("-------message: {:?}", rlp::encode(&vote).to_vec());
 
     let hash_vote = keccak_256(rlp::encode(&vote).as_ref());
     let pks = extract_pks(&proof, validator_list)?;
     let pks = pks.iter().collect::<Vec<_>>();
     let c_pk = PublicKey::from_aggregate(&AggregatePublicKey::aggregate(&pks, true)?);
     let sig = Signature::from_bytes(&proof.signature)?;
+    println!("--------signature: {:?}", proof.signature.to_vec());
```
Comment (KaoImin) [inline axon-tools/src/proof.rs]: Remove these `print` or change them to `log::debug!()`.

**Review by KaoImin — CHANGES_REQUESTED** [review]

File: examples/verify_proof.rs (hunk 1 above)
Comment (wenyuanhust) [reply examples/verify_proof.rs]: if rand is used, the final result won't be fixed, which can not be verified.
this test is used to compare the axon-tools generated proposal hash is the same with axon.

**Review by KaoImin — APPROVED** [review]

**Pull request merged**